### PR TITLE
Final task

### PR DESCRIPTION
**Pull request by kirillnaumenko**

### CODE REVIEW

File: src/test/java/finalTask/Configuration.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    static {
+        JSONParser jsonParser = new JSONParser();
+        try {
+            // Read the JSON file and parse it
+            Object obj = jsonParser.parse(new FileReader(CONFIG_FILE_PATH));
+
+            // Convert the parsed object to a JSONObject
+            config = (JSONObject) obj;
+        } catch (IOException | ParseException e) {
+            e.printStackTrace();
```
Comment (KBugrimCS) [inline src/test/java/finalTask/Configuration.java]: Let's avoid all  e.printStackTrace() methods and replace it to logger

File: src/test/java/finalTask/config.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    "browser": "firefox"
+  },
+  "selenoid": {
+    "url": "localhost",
+    "port": "4444"
+  },
+  "saucelabs": {
+    "url": "https://ondemand.us-west-1.saucelabs.com:443/wd/hub",
+    "username": "KirillNaumenko",
+    "accesskey": "f3201ecc-281c-45e8-ada8-e6ef100b542f"
```
Comment (KBugrimCS) [inline src/test/java/finalTask/config.json:13]: just want to tell you that it is a bad practice to store various credentials in a repo. It is better to store it in environment variables, on the machine where the tests are run, or in the CI/CD tool ,etc

File: src/test/java/finalTask/core/BaseTest.java [hunk 3]
```diff
@@ -1,32 +1,46 @@
...
+            case "selenoid":
+                runner = new SelenoidDriver();
+                break;
+            case "saucelabs":
+                runner = new SauceLabsDriver();
+                break;
+        }
+        driver = runner.setupDriver();
         driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(30));
     }
```
Comment (KBugrimCS) [inline src/test/java/finalTask/core/BaseTest.java]: please move this logic to Webdriver singleton class

File: src/test/java/finalTask/dataModels/AccountModel.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    private String firstName;
+    private String lastName;
+    private String email;
+    private String password;
```
Comment (KBugrimCS) [inline src/test/java/finalTask/dataModels/AccountModel.java]: please format this class

File: src/test/java/finalTask/dataModels/AddressModel.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
+package finalTask.dataModels;
+
+public class AddressModel {
```
Comment (KBugrimCS) [inline src/test/java/finalTask/dataModels/AddressModel.java:3]: and format this class too

File: src/test/java/finalTask/pages/CreateAccountPage.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    @FindBy(id = "lastname")
+    private WebElement lastNameInput;
+    @FindBy(id = "email_address")
+    private WebElement emailInput;
+    @FindBy(id = "password")
+    private WebElement passwordInput;
+    @FindBy(id = "password-confirmation")
+    private WebElement confirmPasswordInput;
+    @FindBy(css = ".primary .action")
+    private WebElement createAccountButton;
```
Comment (KBugrimCS) [inline src/test/java/finalTask/pages/CreateAccountPage.java]: please add blank lines between fields for better readabilitty

**Review by KBugrimCS — CHANGES_REQUESTED** [review]
Good, but I have a couple comments:
1. Where is your driver singleton class?
2. Please format your code according to java code convention
3. Please you logger in your code

File: src/test/java/finalTask/config.json (hunk 2 above)
Comment (kirillnaumenko) [reply src/test/java/finalTask/config.json:13]: yes. But for now i dont have any vault or virtual machine to store credentials somewhere else :)

**Review by KBugrimCS — APPROVED** [review]
looks good

**Pull request merged**